public/app/features/alerting/unified/components/rule-editor/AlertTypeStep.tsx

```diff
@@ -0,0 +1,176 @@
+      if (ruleFormType === RuleFormType.threshold) {
+        return !!ds.meta.alerting;
+      } else {
+        // filter out only rules sources that support ruler and thus can have alerts edited
+        return !!rulesSourcesWithRuler.find(({ id }) => id === ds.id);
+      }
+    },
+    [ruleFormType, rulesSourcesWithRuler]
+  );
+
+  return (
+    <RuleEditorSection stepNo={1} title="Alert type">
+      <Field
+        className={styles.formInput}
+        label="Alert name"
+        error={errors?.name?.message}
+        invalid={!!errors.name?.message}
+      >
+        <Input
+          autoFocus={true}
+          ref={register({ required: { value: true, message: 'Must enter an alert name' } })}
+          name="name"
+        />
+      </Field>
+      <div className={styles.flexRow}>
+        <Field
+          label="Alert type"
+          className={styles.formInput}
+          error={errors.type?.message}
+          invalid={!!errors.type?.message}
+        >
+          <InputControl
+            as={Select}
+            name="type"
+            options={alertTypeOptions}
+            control={control}
+            rules={{
+              required: { value: true, message: 'Please select alert type' },
+            }}
+            onChange={(values: SelectableValue[]) => {
+              const value = values[0]?.value;
+              // when switching to system alerts, null out data source selection if it's not a rules source with ruler
+              if (
+                value === RuleFormType.system &&
+                dataSourceName &&
+                !rulesSourcesWithRuler.find(({ name }) => name === dataSourceName)
+              ) {
+                setValue('dataSourceName', null);
+              }
+              return value;
+            }}
+          />
+        </Field>
+        <Field
+          className={styles.formInput}
+          label="Select data source"
+          error={errors.dataSourceName?.message}
+          invalid={!!errors.dataSourceName?.message}
+        >
+          <InputControl
+            as={DataSourcePicker as React.ComponentType<Omit<DataSourcePickerProps, 'current'>>}
+            valueName="current"
+            filter={dataSourceFilter}
+            name="dataSourceName"
+            noDefault={true}
+            control={control}
+            alerting={true}
+            rules={{
+              required: { value: true, message: 'Please select a data source' },
+            }}
+            onChange={(ds: DataSourceInstanceSettings[]) => {
+              // reset location if switching data sources, as differnet rules source will have different groups and namespaces
+              setValue('location', undefined);
+              return ds[0]?.name ?? null;
+            }}
+          />
+        </Field>
+      </div>
+      {ruleFormType === RuleFormType.system && (
+        <Field
+          label="Group"
+          className={styles.formInput}
+          key={dataSourceName || 'null'}
```

Comment:
why a string null instead of `null`?

Reply:
I actually have no idea what this is for, I don't think we need it. Deleted the `key` prop entirely